package.json

```diff
@@ -1,11 +1,17 @@
 {
-  "name": "baal",
+  "name": "@baal/contracts",
   "version": "1.0.0",
   "description": "Lo, also it is the time of His rain.",
-  "main": "index.js",
+  "main": "dist/src/index.ts",
+  "types": "dist/src/index.d.ts",
   "scripts": {
     "test": "hardhat test --network hardhat",
-    "compile": "TS_NODE_TRANSPILE_ONLY=1 hardhat typechain"
+    "compile": "TS_NODE_TRANSPILE_ONLY=1 hardhat typechain",
+    "build": "npm run build:sol && npm run build:abi && npm run build:ts && npm run build:declarations",
+    "build:ts": "npx tsc --build tsconfig.build.json",
+    "build:abi": "npx hardhat export-abi",
+    "build:sol": "npx hardhat compile",
+    "build:declarations": "shx cp src/types/*.d.ts dist/src/types"
```

Comment:
build scripts to get the final `dist` folder.

we can put other stuff in the `dist` from the contracts if there is a use case

to build all, `npm run build`